modules/expo-bluesky-oauth-client/src/react-native-oauth-client-factory.native.ts

```diff
@@ -0,0 +1,143 @@
+import {Fetch} from '@atproto/fetch'
+import {UniversalIdentityResolver} from '@atproto/identity-resolver'
+import {
+  OAuthAuthorizeOptions,
+  OAuthClientFactory,
+  OAuthResponseMode,
+  OAuthResponseType,
+  Session,
+} from '@atproto/oauth-client'
+import {OAuthClientMetadata} from '@atproto/oauth-client-metadata'
+import {IsomorphicOAuthServerMetadataResolver} from '@atproto/oauth-server-metadata-resolver'
+
+import {ReactNativeCryptoImplementation} from './react-native-crypto-implementation'
+import {DatabaseStore} from './react-native-oauth-database'
+import {RNOAuthDatabase} from './react-native-oauth-database.native'
+
+export type RNOAuthClientOptions = {
+  responseMode?: OAuthResponseMode
+  responseType?: OAuthResponseType
+  clientMetadata: OAuthClientMetadata
+  fetch?: Fetch
+  crypto?: any
+}
+
+export class RNOAuthClientFactory extends OAuthClientFactory {
```

Comment:
Have a look at  `react-native-oauth-client-factory.ts` from the skeleton:
1) It contains a `signIn()` method that shows how this is expected to word
2) It uses a slightly different implementation for the stores that allows proper serialization of the `dpopKey` ;-)